checks/junit.go

```diff
@@ -170,42 +228,48 @@ func (c *JunitChecker) Check(canary v1.Canary, extConfig external.Check) *pkg.Ch
 		}
 		allTestSuite = append(allTestSuite, testSuite...)
 	}
-	//initializing results map with 0 values
-	var failedTests = make(map[string]string)
-	for _, suite := range allTestSuite {
-		for _, test := range suite.Tests {
-			switch test.Status {
-			case junit.StatusFailed:
-				junitStatus.failed++
-				failedTests[suite.Name+"/"+test.Name] = failedTests[test.Message]
-			case junit.StatusPassed:
-				junitStatus.passed++
-			case junit.StatusSkipped:
-				junitStatus.skipped++
-			case junit.StatusError:
-				junitStatus.error++
-			}
-			if test.Status == junit.StatusFailed {
-				failedTests[suite.Name+"/"+test.Name] = failedTests[test.Message]
-			}
-		}
+	var junitResults JunitResult
+	junitResults.Suites = make([]JunitTestSuite, len(allTestSuite))
+	for i, suite := range allTestSuite {
+		// Aggregate results
+		suite.Aggregate()
+
+		junitResults.Passed += suite.Totals.Passed
+		junitResults.Failed += suite.Totals.Failed
+		junitResults.Skipped += suite.Totals.Skipped
+		junitResults.Error += suite.Totals.Error
+		junitResults.Duration += suite.Totals.Duration.Seconds()
+
+		junitResults.Suites[i].Passed = suite.Totals.Passed
+		junitResults.Suites[i].Failed = suite.Totals.Failed
+		junitResults.Suites[i].Skipped = suite.Totals.Skipped
+		junitResults.Suites[i].Error = suite.Totals.Error
+		junitResults.Suites[i].Duration = suite.Totals.Duration.Seconds()
```

Comment:
We still want millisecond resolution e.g. 0.005 for 5ms

```suggestion
		junitResults.Suites[i].Duration = suite.Totals.Duration.Milliseconds() / 1000
```